File: fix_tests.py
import re

with open('src/App.test.tsx', 'r') as f:
    content = f.read()

# For Visuals tab
patterns_visuals = [
    r"getByLabelText\('Image prompt'\)",
    r"getByLabelText\('Active visualizer'\)",
    r"name: 'Visual prompt'",
    r"name: 'Post-processing effects'",
    r"getByTestId\('effect-row"
]

# For Queue tab
patterns_queue = [
    r"name: 'Generation queue'",
    r"getByText\('No generations yet.'\)"
]

blocks = content.split("it('")
for i in range(1, len(blocks)):
    block = blocks[i]
    needs_visuals = any(re.search(p, block) for p in patterns_visuals)
    needs_queue = any(re.search(p, block) for p in patterns_queue)
    
    if needs_visuals or needs_queue:
        # We find the render(<App />); and insert clicks
        if "render(<App />);" in block:
            injection = "render(<App />);\n"
            if needs_visuals:
                injection += "    fireEvent.click(screen.getByRole('button', { name: 'Visual Settings' }));\n"
            if needs_queue:
                injection += "    fireEvent.click(screen.getByRole('button', { name: 'Queue' }));\n"
            
            # Avoid duplicate injections
            if "fireEvent.click(screen.getByRole('button', { name: 'Visual Settings' }));" not in block and "fireEvent.click(screen.getByRole('button', { name: 'Queue' }));" not in block:
                blocks[i] = block.replace("render(<App />);\n", injection)

with open('src/App.test.tsx', 'w') as f:
    f.write("it('".join(blocks))

File: fix_tests.js
const fs = require('fs');

const path = 'src/App.test.tsx';
let content = fs.readFileSync(path, 'utf8');

// The lines we want to inject click for Visual Settings
const visualStrings = [
  "getByLabelText('Image prompt')",
  "getByLabelText('Active visualizer')",
  "getByRole('region', {\n      name: 'Visual prompt'",
  "getByRole('region', {\n        name: 'Visual prompt'",
  "getByRole('group', {\n      name: 'Post-processing effects'",
  "getByRole('checkbox'",
  "getByTestId('effect-row"
];

const blocks = content.split("it('");
for (let i = 1; i < blocks.length; i++) {
  if (visualStrings.some(s => blocks[i].includes(s))) {
    // Inject tab switch after render(<App />);
    if (!blocks[i].includes("fireEvent.click(screen.getByRole('button', { name: 'Visual Settings' }));")) {
      blocks[i] = blocks[i].replace(
        "render(<App />);", 
        "render(<App />);\n    fireEvent.click(screen.getByRole('button', { name: 'Visual Settings' }));"
      );
    }
  }
}

content = blocks.join("it('");

// Fix queue tests
const queueStrings = [
  "getByText('No generations yet')",
  "getByRole('region', { name: 'Generation queue' })"
];

const blocksQ = content.split("it('");
for (let i = 1; i < blocksQ.length; i++) {
  if (queueStrings.some(s => blocksQ[i].includes(s))) {
    if (!blocksQ[i].includes("fireEvent.click(screen.getByRole('button', { name: 'Queue' }));")) {
      blocksQ[i] = blocksQ[i].replace(
        "render(<App />);", 
        "render(<App />);\n    fireEvent.click(screen.getByRole('button', { name: 'Queue' }));"
      );
    }
  }
}
content = blocksQ.join("it('");

fs.writeFileSync(path, content, 'utf8');
